x-pack/plugins/cloud_security_posture/common/utils/helpers.test.ts

```diff
@@ -185,5 +185,37 @@ describe('test helper methods', () => {
         shared_credential_file: { value: undefined, type: 'text' },
       });
     });
+
+    it('cleans unused gcp credential methods, when using credentials-file method ', () => {
```

Comment:
Can you add tests for when gcp.credentials.type and aws.credentials.type are null/undefined? I should have included that in my changes as it resulted in a bug.